src/backend/columnar/columnar_reader.c

```diff
@@ -288,12 +304,19 @@ BeginStripeRead(StripeMetadata *stripeMetadata, Relation rel, TupleDesc tupleDes
 
 
 /*
- * EndStripeRead finishes a stripe read.
+ * EndStripeRead finishes current stripe read and increments currentStripe for
+ * next one.
  */
 static void
-EndStripeRead(StripeReadState *stripeReadState)
+EndStripeRead(ColumnarReadState *readState)
 {
-	pfree(stripeReadState);
+	readState->chunkGroupsFiltered +=
+		readState->stripeReadState->chunkGroupsFiltered;
+	pfree(readState->stripeReadState);
+	readState->stripeReadState = NULL;
+	MemoryContextReset(readState->stripeReadContext);
+
+	readState->currentStripe++;
 }
```

Comment:
I'd prefer to keep destruction of the stripeReadState separate from advancing to a new stripe. We could keep EndStripeRead as-is, and create a new function AdvanceStripe or something?

Reply:
makes sense, what about b007d242300501afee9dcc9a572901ad5154450b ?